tests/countries/test_sierra_leone.py

```diff
@@ -0,0 +1,194 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS.md file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.countries import SierraLeone, SL, SLE
+from tests.common import CommonCountryTests
+
+
+class TestSierraLeone(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        years = range(1961, 2050)
+        super().setUpClass(SierraLeone, years=years, years_non_observed=years)
+
+    def test_country_aliases(self):
+        self.assertAliases(SierraLeone, SL, SLE)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(SierraLeone(years=1960))
+
+    def test_new_years_day(self):
+        self.assertHolidayName(
+            "New Year's Day",
+            (f"{year}-01-01" for year in range(1961, 2050)),
+        )
```

Comment:
```suggestion
        self.assertHolidayName("New Year's Day", (f"{year}-01-01" for year in range(1962, 2050)))
```
And same for rest similar cases. Also need to add check for "observed" for all holidays where it can be (as for Independence Day).

Reply:
> Also need to add check for "observed" for all holidays where it can be (as for Independence Day).

👆